src/store.ts

```diff
@@ -0,0 +1,163 @@
+import Ain from '@ainblockchain/ain-js';
+import axios from 'axios';
+import stringify = require('fast-json-stable-stringify');
+import { HttpMethod, PurchaseHistory, StoreItem, StorePurchaseParams, UserItem } from './types';
+import { buildData } from './util';
+
+export default class Store {
+  private baseUrl: string;
+  public ain: Ain;
+
+  constructor(baseUrl: string, ain: Ain) {
+    this.baseUrl = `${baseUrl}/store`;
+    this.ain = ain;
+  }
+
+  setBaseUrl(baseUrl: string) {
+    this.baseUrl = `${baseUrl}/store`;
+  }
+
+  signData(data: any) {
```

Comment:
maybe the repeated methods and member variables like `signData()` and `ain` can be moved to a base class e.g. AinftBase?

Reply:
Done! Add AinftBase class as base class and moved `signDate()`, `ain`, `setBaseUrl` to base class.